### PR TITLE
Issue #548 Align XML element, attribute and XSD type names

### PR DESCRIPTION
**Pull request by ljacomet**
... on words separated by dash

### CODE REVIEW

**Comment by ljacomet** [comment]
Well, now the PR has all it needs, not just the XSD changes ;-)

File: xml/src/main/resources/ehcache-core.xsd [hunk 1]
```diff
@@ -221,13 +221,13 @@
...
-          <xs:attribute type="xs:positiveInteger" name="concurrency" default="1" use="optional"/>
-          <xs:attribute type="xs:positiveInteger" name="size" default="2147483647" use="optional"/>
-          <xs:attribute type="xs:boolean" name="coalesce" default="false" use="optional"/>
-          <xs:attribute name="ratelimitpersecond" type="xs:positiveInteger" default="2147483647" use="optional"/>
-          <xs:attribute name="minWriteDelay" type="xs:nonNegativeInteger" default="0" use="optional"/>
-          <xs:attribute name="maxWriteDelay" type="xs:positiveInteger" default="2147483647" use="optional"/>
-          <xs:attribute name="batchsize" type="xs:positiveInteger" default="1" use="optional"/>
+          <xs:attribute name="concurrency" type="xs:positiveInteger" default="1" use="optional"/>
+          <xs:attribute name="size" type="xs:positiveInteger" default="2147483647" use="optional"/>
+          <xs:attribute name="coalesce" type="xs:boolean" default="false" use="optional"/>
```
Comment (alexsnaps) [inline xml/src/main/resources/ehcache-core.xsd]: Why this flip?

Comment (ljacomet) [inline xml/src/main/resources/ehcache-core.xsd]: I could put it in a separate commit - main reason is readability, I actually missed these when I did work on #534

Comment (ljacomet) [inline xml/src/main/resources/ehcache-core.xsd]: Done, the three lines that are a cosmetic change have been moved to an isolated commit

File: xml/src/main/resources/ehcache-core.xsd [hunk 2]
```diff
@@ -37,7 +37,7 @@
             </xs:documentation>
           </xs:annotation>
         </xs:element>
-        <xs:element name="cache-template" type="ehcache:cacheTemplate-type">
+        <xs:element name="cache-template" type="ehcache:cache-template-type">
           <xs:annotation>
             <xs:documentation xml:lang="en">
               A cache template definition that can be reference in &lt;cache&gt; definitions using the 'usesTemplate'
```
Comment (chrisdennis) [inline xml/src/main/resources/ehcache-core.xsd]: `uses-template`

File: xml/src/main/resources/ehcache-core.xsd [hunk 3]
```diff
@@ -112,9 +112,9 @@
...
   </xs:complexType>
 
-  <xs:complexType name="cacheTemplate-type">
+  <xs:complexType name="cache-template-type">
     <xs:complexContent>
-      <xs:extension base="ehcache:baseCache-type">
+      <xs:extension base="ehcache:base-cache-type">
         <xs:attribute name="name" type="xs:ID" use="required">
           <xs:annotation>
             <xs:documentation xml:lang="en">
```
Comment (chrisdennis) [inline xml/src/main/resources/ehcache-core.xsd]: `uses-template` again

Comment (ljacomet) [inline xml/src/main/resources/ehcache-core.xsd]: Fixed
